tooling/kube_tool.rb

```diff
@@ -29,9 +30,12 @@
     options[:container_runtime] = container_runtime;
   end
 
-  opts.on('-c', '--cni-provider cni-provider', 'the networking provider to use') do |cni_provider|
+  opts.on('-c', '--cni-provider cni-provider', 'the networking provider to use, flannel, weave, calico or cilium are supported') do |cni_provider|
     options[:cni_provider] = cni_provider;
   end
+  opts.on('-cv', '--cni-provider-version [cni_provider_version]', 'the networking provider version to use, calico and cilium will use this to reference the correct deployment downloadlink') do |cni_provider_version|
```

Comment:
Apparently the Ruby OptionParser ( https://ruby-doc.org/stdlib-2.2.2/libdoc/optparse/rdoc/OptionParser.html ) gets a bit confused because of the -c for cni-provider, therefore I think you should either rename the -cv to a simple character or just keep it as --cni-provider-version (without the square bracket value).

Reply:
Docs said setting the square brackets marks the parameter as optional. I will have a look at it again